spec/javascripts/NewService/formElements/BackendApiSelect.spec.jsx

```diff
@@ -0,0 +1,40 @@
+// @flow
+
+import React from 'react'
+import {act} from 'react-dom/test-utils'
+import Enzyme, {mount, shallow, render} from 'enzyme'
+import Adapter from 'enzyme-adapter-react-16'
+
+import {BackendApiSelect} from 'NewService/components/FormElements'
+import * as utils from 'utilities/utils'
+import { BASE_PATH } from 'NewService'
+
+Enzyme.configure({adapter: new Adapter()})
+
+const props = {
+  backendApis: []
+}
+
+it('should render itself', () => {
+  const wrapper = shallow(<BackendApiSelect {...props} />)
+  expect(wrapper.find('#service_act_as_product_input').exists()).toEqual(true)
+})
+
+it('should render Act as Product checkbox', () => {
+  const wrapper = shallow(<BackendApiSelect {...props} />)
+  expect(wrapper.find('input[name="service[act_as_product]"]').exists()).toEqual(true)
+})
+
+it('should not render the field to select the Backend Api by default', () => {
```

Comment:
Maybe "should not render select element when Act as Product checkbox is unchecked" ?